overlord/snapstate/snapmgr.go

```diff
@@ -234,6 +234,13 @@ func ComponentSetupsForTask(t *state.Task) ([]*ComponentSetup, error) {
 	case t.Has("component-setup") || t.Has("component-setup-task"):
 		// task comes from a singular component installation for an already
 		// installed snap
+		// if the task isn't being done right now (we may be undoing it), then
+		// we shouldn't consider the new component when setting up profiles.
+		if t.Status() != state.DoingStatus {
```

Comment:
it's a bit strange to put the logic here, we do not do this way for any other undo as far as I know